### DISCUSSION TITLE
ROM decision guidance re: secure & VNDK

### DISCUSSION THREAD

**Discussion by DanielR-I** [phhusson/treble_experimentations#2494] [Q&A] [answered]
I've decided to use a treble rom to try "degoogle" my **Motorola G Play (2021)** (XT2093-4 GUAMNA) phone... sans GAPPS

**Treble Info** tool tells me that the **required image** for my phone is _**system-`arm64`-`ab`.img.xz**_ which I think it based on the other information it provided:
- **Treble and VNDK:** supports project treble, using VNDK, version 30.0
- **System as Root:** supported, use `A/B`
- **Architechture:** `ARM64`
- **Seemless Upgrades:** supported
- **Dynamic Partitions:** Are used by my device... "high-level partitions" flashed via fastbootd

Questions this raised once I started looking at the ROM release files (which remained unanswered by the wiki):
1. If my phone supports treble using `VNDK`, why wasn't a ROM like "system-squeak-arm64-ab-**`vndk`lite**-vanilla.img.xz" that includes it in it's name suggested?  Or maybe a better question would be... Is there a difference between **VNDK** and **VNDKlite**, and is **VNDK** implied?
2. Not wanting GAPPS and feeling confident I can load F-DROID and other FOSS apps myself afterward, I'm thinking `vanilla` is the way to go for me... but there are some `vanilla`'s that are also "**secure**" like  
"system-squeak-arm64-ab-vndklite-`vanilla`-**secure**.img.xz"... why might i want or not want a "**secure**" ROM?  It appears "**secure**" is only an option with a **VNDKlite** included ROM.
3. Hmm, I might be making an assumption about **FOSS**... I noticed about 250MB size difference between the **vanilla** and **floss** ROM's... is that just the 15 apps listed in the wiki, or might there be more differences?  Is a GAPPS alternative like MicroG in play?

The answers to my questions should help me decide between:
- system-squeak-arm64-ab-`vanilla`.img.xz (closest to the **Treble Info** recommendation)
- system-squeak-arm64-ab-`vndklite`-`vanilla`.img.xz
- system-squeak-arm64-ab-`vndklite`-`vanilla`-`secure`.img.xz
- system-squeak-arm64-ab-`floss`.img.xz
- system-squeak-arm64-ab-`vndklite`-`floss`.img.xz
- system-squeak-arm64-ab-`vndklite`-`floss`-`secure`.img.xz

------
Additional Questions inspired by **Treble Info**:

A) **Seemless Upgrades:** This I'm sure is something very different than "automatic updates", but raises the question... After flashing the ROM, is there an automation for "system" updates, or is the idea to flash an updated system ROM down the road? If subsequent flashing will be required, will that wipe my user data (installed apps and their data)?

B) **Dynamic Partitions:** Which are the "high-level partitions"? Is System one of them? Oddly/conflictingly... according to https://forum.xda-developers.com/t/guide-using-fastboot-exe-with-motorola-devices.4042039/post-81656729 fastbootd is not used by my device...
```
$ adb reboot bootloader
$ fastboot getvar is-userspace
is-userspace: no
Finished. Total time: 0.000s
```
... which is weird, because the instructions I was going to follow for my phone @  https://forum.xda-developers.com/t/any-custom-roms-out-there-for-moto-g-play-2021-guamna.4260871/post-86389207 do mention it in **Step 7**.

I think I'll trust **Treble Info** and the flash instructions I found and forget the `$ fastboot getvar is-userspace` command, because i definitely can load into a **fastbootd** screen on this phone...
```
$ adb reboot bootloader
$ fastboot reboot fastboot
Rebooting into fastboot                            OKAY [  0.002s]
< waiting for any device >
Finished. Total time: 8.621s
```

**Comment by phhusson** [comment, marked as answer]
"Seamless upgrades" is pretty much irrelevant for GSIs. GSI that implement OTA have it "seamless" no matter the device (but it works only on dynamic partition devices)
To know which partitions are dynamic you can look at `fastboot getvar all`

If device isn't vndklite (treble info would say if it was), you can use both vndklite and non-vndklite images

Secure fakes various properties to make it look like you're running stock OEM ROM, and removes any way to get root (which is annoying)

**Reply by DanielR-I** [reply]
Thank you for the reply!  So far it sounds like I should not choose **vndklite** and **secure** ROMs.

Do the **floss** versions just preload the 15 apps (f-droid, etc) or do they also contain a GAPPS replacement like microg?  

Happy Holidays!

**Reply by phhusson** [reply]
floss includes microg. I don't really understand your other questions. microg is opensource, by default it discusses a tiny bit (nothing compared to original google apps) with google servers, but you can disable it in its settings

**Reply by DanielR-I** [reply]
Thank you for your reply.  By GAPPS I was referring to the proprietary background google services that plug a phone into the google ecosystem 24/7... so you've definitely answered my question.

It's nice to know microg is included in floss and can be disabled in the settings.

**Reply by DanielR-I** [reply]
As soon as my developer option for **OEM Unlocking** comes available I'll move forward with the **system-squeak-arm64-ab-floss.img.xz ROM** and initially disable MicroG so I can get a sense for whether anything I use really requires it.
